spring-expression/src/test/java/org/springframework/expression/spel/SpelCompilationCoverageTests.java

```diff
@@ -2277,31 +2285,33 @@ public static String join(String...strings) {
 
 	@Test
 	void compiledExpressionShouldWorkWhenUsingCustomFunctionWithVarargs() throws Exception {
-		StandardEvaluationContext context = null;
+		StandardEvaluationContext context;
 
-		// Here the target method takes Object... and we are passing a string
+		// single string argument
 		expression = parser.parseExpression("#doFormat('hey %s', 'there')");
 		context = new StandardEvaluationContext();
 		context.registerFunction("doFormat",
 				DelegatingStringFormat.class.getDeclaredMethod("format", String.class, Object[].class));
 		((SpelExpression) expression).setEvaluationContext(context);
 
 		assertThat(expression.getValue(String.class)).isEqualTo("hey there");
-		assertThat(((SpelNodeImpl) ((SpelExpression) expression).getAST()).isCompilable()).isTrue();
+		assertThat(((SpelExpression) expression).getAST().isCompilable()).isTrue();
```

Comment:
I see that you have made this change numerous times within this PR.

As I mentioned in a previous comment, we should avoid including refactorings like this in a PR, since these changes make it much more challenging to follow what the PR is trying to achieve.